force-app/main/default/classes/NextStepBatch.cls

```diff
@@ -22,36 +22,52 @@ public with sharing class NextStepBatch implements Database.Batchable<sObject>{
 
     public Database.QueryLocator start(Database.BatchableContext bc){
         // Query for all accounts and return the query locator.
-        return null; // Update null with the query locator
+        return Database.getQueryLocator([SELECT Id, Name, Next_Step__c FROM Account LIMIT 5]); // Update null with the query locator
     }
     
     public void execute(Database.BatchableContext bc, List<Account> scope){
         final String BATCH_NEXT_STEP = 'Batching the next meetings together.';
 
         // Update the Next Step field on each record in the batch scope.
-
         // Loop through the scope account records
-
-            // Set the Next Step field to 'Batching the next meetings together.'
-
+        // Set the Next Step field to 'Batching the next meetings together.'
         // Update the account records in this batch scope
+        for (Account acc : scope) {
+            acc.Next_Step__c = BATCH_NEXT_STEP;
+        }
+        try {
+            update scope;
+        } catch (DmlException e) {
+            System.debug('Account Update Error: ' + e.getMessage()); 
+        }
         
         // Query for all opportunities related to the accounts in this batch scope
-
         // Loop through the opportunity records
-
-            // Set the Next Step field to 'Batching the next meetings together.'
-
+        // Set the Next Step field to 'Batching the next meetings together.'
         // Update the opportunity records
 
+        List<Opportunity> scopeOpps = [SELECT Id, Name, Description, NextStep FROM Opportunity LIMIT 5];
+        for (Opportunity opp : scopeOpps) {
+            opp.NextStep = BATCH_NEXT_STEP;
+        }
+       try {
+            update scopeOpps;
+       } catch (DmlException e) {
+        System.debug('Oppty Update Error: ' + e.getMessage());
+       } 
     }
     
     public void finish(Database.BatchableContext bc){
         final List<User> USERS = [SELECT Id, Name, Email FROM User WHERE IsActive = true AND (NOT Email LIKE 'autoproc%')]; // DO NOT CHANGE
 
         // Get the email addresses of all active users
-
-            // Pass the set of active user emails to the sendEmailToActiveUsers method
+        // Pass the set of active user emails to the sendEmailToActiveUsers metho
+        Set<String> emailSet = new Set<String>();
+            for (User usr : USERS) {
+                emailSet.add(usr.Email);
+            }
+            sendEmailToActiveUsers(emailSet);
```

Comment:
Great work. I would just suggest checking that emailSet contains values before calling the send email method